### ISSUE TITLE
Add hyperloglog functions to bigquery and snowflake

### ISSUE THREAD

**Issue by lloydtabb** [malloydata/malloy#2054]
Hyperloglog is implemented in Presto and Trino.  We need to add it to BigQuery and Snowflake and write the documentation.

See the pull requests below for the documentation
https://github.com/malloydata/malloy/pull/2053
https://github.com/malloydata/malloy/pull/2055

**Comment by anagrd-ai** [comment]
Added a new issue in the docs repo to track documentation, as that can be done in parallel:
https://github.com/malloydata/malloydata.github.io/issues/207

This issue will now track just the coding work.

For all the details, see the design doc: https://docs.google.com/document/d/1acWaGvlPjFrdAh-gJ1bvulL8d2htaARqjX7a5tqWeL8/edit?tab=t.0
